app/code/Magento/Checkout/Test/Unit/Plugin/ResetQuoteAddressesTest.php

```diff
@@ -0,0 +1,172 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Checkout\Test\Unit\Plugin;
+
+use Magento\Checkout\Plugin\Model\Quote\ResetQuoteAddresses;
+use Magento\Quote\Api\Data\CartExtensionInterface;
+use Magento\Quote\Model\Quote;
+use Magento\Quote\Model\Quote\Address;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Class ResetQuoteAddressesTest
+ *
+ * Test of clearing quote addresses after all items were removed.
+ */
+class ResetQuoteAddressesTest extends TestCase
+{
+    /**
+     * @var ResetQuoteAddresses
+     */
+    private $plugin;
+
+    /**
+     * @var Quote|MockObject
+     */
+    private $quoteMock;
+
+    /**
+     * @var CartExtensionInterface|MockObject
+     */
+    private $extensionAttributesMock;
+
+    /**
+     * Set Up
+     */
+    protected function setUp()
+    {
+        $this->quoteMock = $this->createPartialMock(Quote::class, [
+            'getAllAddresses',
+            'getAllVisibleItems',
+            'removeAddress',
+            'getExtensionAttributes',
+            'isVirtual',
+        ]);
+        $this->extensionAttributesMock = $this->getMockBuilder(CartExtensionInterface::class)
+            ->setMethods(
+                [
+                    'getShippingAssignments',
+                    'setShippingAssignments'
+                ]
+            )
+            ->getMockForAbstractClass();
+
+        $this->plugin = new ResetQuoteAddresses();
+    }
+
+    /**
+     * Test removing the addresses from a non empty quote
+     */
+    public function testRemovingTheAddressesFromNonEmptyQuote()
+    {
+        $quoteVisibleItems = [1, 2];
+
+        $this->quoteMock->expects($this->any())
+            ->method('getAllVisibleItems')
+            ->will($this->returnValue($quoteVisibleItems));
+        $this->quoteMock->expects($this->never())
+            ->method('getAllAddresses')
+            ->willReturnSelf();
+
+        $this->plugin->afterRemoveItem($this->quoteMock, $this->quoteMock, 1);
+    }
+
+    /**
+     * Test clearing the addresses from an empty quote
+     *
+     * @dataProvider quoteDataProvider
+     * @param bool $isVirtualQuote
+     * @param bool $quoteHasAddresses
+     * @param $extensionAttributes
+     */
+    public function testClearingTheAddressesFromEmptyQuote(
+        bool $isVirtualQuote,
+        bool $quoteHasAddresses,
+        $extensionAttributes
+    ) {
+        $quoteVisibleItems = [];
+
+        $this->quoteMock->expects($this->any())
+            ->method('getAllVisibleItems')
+            ->will($this->returnValue($quoteVisibleItems));
+
+        if ($quoteHasAddresses) {
+            $address = $this->createPartialMock(Address::class, ['getId']);
+
+            $address->expects($this->any())
+                ->method('getId')
+                ->willReturn(1);
```

Comment:
Try to avoid Magic Numbers - use `private const STUB_ADDRESS_ID` instead.

Reply:
Updated. 👌